app/test/specs/configProfileControllerSpec.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+(describe('Test ConfigProfileController', function() {
+    var configCtrl, httpBackend, deffered, scope, userService, createCrtl;
+    var splab = {
+        name: 'SPLAB',
+        key: '987654321' 
+    };
+
+    var user = {
+        name: 'Maiana',
+        cpf: '121.445.044-07',
+        email: 'maiana.brito@ccc.ufcg.edu.br',
+        institutions: [splab]
+    };
+
+    var newUser = {
+        name: 'Maiana Brito',
+        cpf: '121.115.044-07',
+        email: 'maiana.brito@ccc.ufcg.edu.br',
+        institutions: [splab]
+    };
+
+    beforeEach(module('app'));
+
+    beforeEach(inject(function($controller, $httpBackend, $rootScope, $q, UserService) {
+        httpBackend = $httpBackend;
+        scope = $rootScope.$new();
+        deffered = $q.defer();
+        userService = UserService;
+        createCrtl = function() {
+            return $controller('ConfigProfileController', {scope: scope});
+        };
+        httpBackend.expect('GET', '/api/user').respond(user);
+        configCtrl = createCrtl();
+        httpBackend.flush();   
+    }));
+
+    afterEach(function() {
+        httpBackend.verifyNoOutstandingExpectation();
+        httpBackend.verifyNoOutstandingRequest();
+    });
+
+
+    it("User isn't valid", inject(function($mdToast){
+        spyOn($mdToast, 'show');
+
+        var userInvalid = {
+            name: 'Maiana Brito',
+            cpf: '',
+            email: 'maiana.brito@ccc.ufcg.edu.br',
+            institutions: [splab]
+        };
+        configCtrl.newUser = new User(userInvalid);
+        configCtrl.finish();
+
+        expect($mdToast.show).toHaveBeenCalled(); 
+    }));
+
+    it('Spy save user in success case', function() {
+        spyOn(userService, 'save').and.returnValue(deffered.promise);
+        deffered.resolve();
+        scope.$apply();
+        configCtrl.finish();
+        expect(userService.save).toHaveBeenCalled();
```

Comment:
pq não fazer um único teste que verifica se o save, o state.go e o usuário foram modificados? É estranho fazer vários it para o msm cenário de teste.

O cenário ( o it ) só muda quando vc quer testar um estado diferente, no caso de sucesso, do usuário ser inválido, de o save dá erro, etc.